doc/subsystems/chaos_pp/chaos_pp.rst

```diff
@@ -0,0 +1,916 @@
+..
+    Copyright (c) 2018 Bobby Noelte
+    SPDX-License-Identifier: Apache-2.0
+
+.. _chaos_pp:
+
+Chaos Preprocessor Library
+##########################
+
+Description
+***********
+
+The Chaos Preprocessor Library is a generative metaprogramming framework for C and C++ created by Paul Mensonides. It is a header-only library.
+
+The Chaos Preprocessor Library documentation is provided as HTML files. These are only accessable in the local GIT repository. As a work-around use the [download] link to get the same file from a remote documentation site. Links will not work in the download file.
+
+Concepts
+********
+
+Paul Mensonides:
+
+  "Chaos" does mean something...  It is a reference to chaos theory where a relatively simple set of rules
+  and initial conditions can produce a complex result.
+
+Active Arguments
+================
+
+An active argument is an argument that expands each time that it is scanned by the preprocessor.
+
+ * `Active Arguments <file:../../../../../ext/lib/preprocessor/chaos/built-docs/active-arguments.html>`_
```

Comment:
Hmm... we need to figure out a better way to link to this documentation...

Reply:
Would it be possible to copy the html files to the sphinx html output directory during make htmldocs? Sorry there is no offical web site that can be referenced.

Reply:
I'm not feeling good about using the Zephyr project as the home for documenting an (unrelated) external component (Chaos preprocessor library).  I'll let @nashif advise how to proceed here.